src/pages/characters/characters.ts

```diff
@@ -0,0 +1,119 @@
+
+export class CharactersPage implements IPage {
+  data: any;
+  restService: RestService;
+  numberOfCurrentCharacterPage: number;
+  itemsPerPage: number;
+  scrollObserver: IntersectionObserver;
+  $lastCard: Element; // Last card in page. Used to determine when trigger the data request
+  numberOfCharacterPages: number;
+
+  constructor() {
+    this.restService = window['appContext'].getService('restService');
+    this.data = {};
+    this.data['characters'] = [];
+    this.numberOfCurrentCharacterPage = 0;
+    this.itemsPerPage = 20; // TODO: May be would be fine to move it to a constant
+    this.numberOfCharacterPages = 0;
+  }
+
+  preRender() {
+    return this.getCharacters();
+  }
+
+  render() {
+    const cards = this.data.characters.map(
+      (character: ICharacter) => `<app-card character-name="${character.name}" img-src="${character.image}" character-id="${character.id}"></app-card>`
+    ).join('');
+
+    return `
+      <div class="app-page page-characters">
+        <div class="cards">
+          ${cards}
+        </div>
+      </div>
+    `;
+  }
+
+  setEventHandlers() {
+    this.setScrollHandler();
+    // Add here more handlers
+  }
+
+  setScrollHandler() {
+    if (this.numberOfCurrentCharacterPage >= this.numberOfCharacterPages) {
+      this.scrollObserver.unobserve(this.$lastCard);
+      return;
+    }
+
+    const $cards = document.querySelectorAll('app-card');
```

Comment:
Please leave all vars definitions at the start of the method

Reply:
I put it like this thinking about not using memory when it's not necessary, but suddenly it's an over-optimization. 👍